File: eolymp/87-robot/main.c
#include <stdio.h>

int main()
{
    char in;
    int leftmost = 0, rightmost = 0, ind = 0;

    while (scanf("%c", &in) == 1 && (in == 'L' || in == 'R' || in == 'S'))
    {
        if (in == 'R')
        {
            ++ind;
            if (ind > rightmost)
                rightmost = ind;
        }
        else if (in == 'L')
        {
            --ind;
            if (ind < leftmost)
                leftmost = ind;
        }
    }

    printf("%d", rightmost - leftmost + 1);
    return 0;
}

File: eolymp/87-robot/main.py
seq = input()

lefmost = rightmost = ind = 0

for cmd in seq:
    if cmd == 'R':
        ind += 1
        rightmost = max(rightmost, ind)
    elif cmd == 'L':
        ind -= 1
        lefmost = min(lefmost, ind)
    
print(rightmost - lefmost + 1)